### PR TITLE
Fix inaccuracies and mistakes in comments (especially phpdoc comments)

### PR DESCRIPTION
**Pull request by braders**
Overview
----------------------------------------
Fix inaccuracies and mistakes in comments (especially phpdoc comments).

A lot of the docblock areas in the CiviCRM codebase could do with improvement. This tackles some low-hanging fruit:
 - Inaccurate type definitions
 - Spelling mistakes
 - Inaccurate descriptions

A lot of the offending comments were generated and added in bulk in https://github.com/civicrm/civicrm-core/commit/a0ee39412f184b0b9406094847554851ca2869f2, and so it does make sense that the accuracy may be lacking.

Technical Details
----------------------------------------
In a handful of cases there are examples like this:
```
  /**
   * @param $refTable
   * @param $refKey
   * @param string $targetTable
   * @param string $targetKey
   * @param null $refTypeColumn
   */
  public function __construct($refTable, $refKey, $targetTable = null, $targetKey = 'id', $refTypeColumn = NULL) {
```

In these cases I've made the `* @param string $targetTable` match the expected input (`string`), and not document the `null` default value. There is an argument that either the default value in these cases should become and empty string, or the comment should be updated to `* @param string|null $targetTable`. However, it's not clear in these cases that `null` is actually a valid parameter for the function to work correctly.

### CODE REVIEW

File: CRM/Financial/Form/Export.php [hunk 1]
```diff
@@ -31,7 +31,7 @@ class CRM_Financial_Form_Export extends CRM_Core_Form {
 
   /**
    * Financial batch ids.
-   * @var array
+   * @var string (comma-separated array)
```
Comment (colemanw) [inline CRM/Financial/Form/Export.php:34]: This change appears to be incorrect. The variable is initialized as an array, not a string.

**Comment by colemanw** [comment]
@braders code cleanup is always welcome and appreciated! CiviCRM was originally written before IDEs on PHP version 4, so things have progressed quite a bit since then, and it's great to keep docblocks up-to-date. I do that too as much as I can.

The testbot flagged 2 issues, the first is the same as my comment above on `CRM_Financial_Form_Export`. The second is a bunch of trailing whitespace you left in `CRM_Extension_Info` - you ought to be able to configure your IDE to automatically delete spaces at the end of every line. I use PHPStorm and it does that for me so I don't have to think about it.

File: CRM/Financial/Form/Export.php (hunk 1 above)
Comment (braders) [reply CRM/Financial/Form/Export.php:34]: I think this is actually a string, based on how the property is used:

 - Whenever the propery is used it's preceeded by an `explode(',', $_GET['batch_id'])`, which indicates a string is expected.
 - In `CRM_Batch_BAO_Batch::getBatchNames`, `$this->_batchIds` is passed in as the first agument which then gets fed directly into a `WHERE id IN(...)` database query.

So I think this is a string, which gets exploded whenever it needs to be used as a true PHP array.

**Comment by colemanw** [comment]
Makes sense, thanks again!

**Pull request merged**